### ISSUE TITLE
improved reporting groups

### ISSUE THREAD

**Issue by ethangodt** [grawk/nemo-runner#7]
In the latest  published version reports are not grouped when tests are run in parallel.
- when split on profile `nemo-runner` produces `nemo-report.html` for each profile one after the other as each parallel line of tests complete (each one saving over the last)
- when split on `"parallel":"file"` reports are given parameterized names like `/nemo-report_@tag@_tests!buy-basic!buy-card.json` (one test result per report file)

I'm interested if we can group all test reports regardless of series or parallel scheme. We'd be looking for a single `nemo-report.html` generated in any case that would read something like as an example:
```
{profile1} /tests/test1
{profile2} /tests/test1
{profile1} /tests/test2
{profile2} /tests/test2
```
Having more than one file makes it harder for us to assess the pass/fail rate of any given run. Happy to help!

**Comment by grawk** [comment]
This is a bug in the current release. I'll publish the patch and it should fix the "overwrite" problem. But summarizing parallel reports into one is a separate issue.

**Comment by grawk** [comment]
which version did you base this on? `v1.0.1` should resolve the test file overwrite issue

**Comment by ethangodt** [comment]
@grawk it's possible I saw it and made note, then updated. I'm using `v1.0.1` now. Let me check again.

**Comment by ethangodt** [comment]
@grawk it's still happening so I will update this issue, and I've opened another one [here](https://github.com/paypal/nemo-runner/issues/10) for the bug

**Comment by grawk** [comment]
If both issues are dealing with the same root cause, let's pick one...

I see now that I'm not appending anything to the report name in the case of profile. In my examples, each profile definition defined its own unique report name. We could rightly modify that behavior though. Probably that's a good idea, to cut down on configuration.

**Comment by ethangodt** [comment]
@grawk that will fix the overriding issue, but will leave us with more reports than we have now — what are your thoughts about allowing reports to be merged into one `nemo-report.html`?

**Comment by grawk** [comment]
It's not so much about "allowing" reports to be merged as it is about _how_ would we do that? because right now you get one report per mocha instance. and there could be overlapping tests in the reports. i.e. multiple grep commands could grab overlapping tests. multiple profiles could do the same. but the context of the mocha instance could differentiate one "testA" from another. so the separate reports allow for that. it's not clear how you would merge them into one without a reconciliation step after all the runs complete.

**Comment by grawk** [comment]
i believe if you are using the mocha-jenkins-reporter, multiple reports would be summarized at a high level by the Jenkins junit plugin. so that functionality is "external" to this module in at least the Jenkins case. I'm open to ideas of how to do that within the context of nemo-runner itself, but my feeling is that it would be a fair amount of effort.

**Comment by ethangodt** [comment]
@grawk Interesting. I don't know much about how mocha generates reports. I agree that a reconciliation step does not sound great (buggy and beyond the scope of `nemo-runner`).

If the same test were to be matched by two separate greps, I don't think I'd mind it showing up twice in the aggregate report (assuming this doesn't bug out `mochawesome` or any other tool). This is a little ahead of myself though, not even sure how we'd get an aggregate report...

...let me take a look at the Jenkins plugin, and make sure I really have a need here. My main concern is that for every test run (the goal at some point will be to run potentially 100+ tests on every PR using our app's mocks mode) we'll need to click on 100+ test links to see pass/fail for 1 test. Thanks for response.

**Comment by grawk** [comment]
i can see how this would be a usability problem. it would probably be straightforward (if time consuming) to come up with a report aggregation approach that would resolve this. Although, before digging into it, I can't say that it will be compatible with every possible mocha reporter.

**Comment by ethangodt** [comment]
@grawk fyi, I was planning on just writing some script that would merge the `.json` files that were output, and then generate new html. We would run it as a totally separate step from `nemo-runner`.

Curious if you had an initial thought here about how you would do this.

Also, yeah we use `mochawesome` and probably will for foreseeable future. Very understandable that the aggregation wouldn't work for any reporter.

**Comment by grawk** [comment]
This PR: https://github.com/paypal/nemo-runner/pull/25 will produce much better report separation. In addition it provides top level JSON summaries for each nemo-runner run.
